study1/week4/FloodFill.java

```diff
@@ -0,0 +1,45 @@
+class Solution {
+    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
+        int rowEnd = image.length;
+        int columnEnd = image[0].length;
+
+        if(image[sr][sc] == color) {
+            return image;
+        }
+
+        int value = image[sr][sc];
+        image[sr][sc] = color;
+        Queue<int[]> queue = new LinkedList<>();
+        queue.offer(new int[]{sr, sc});
+
+        while (!queue.isEmpty()) {
+            int[] current = queue.poll();
+            int x = current[0];
+            int y = current[1];
+            int[][] directions = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
```

Comment:
I think this better be a static variable. because it'll be initialized every time you execute this while loop in each iteration. 
Anyways, as I explained in other problems, this comparison is unnecessarily inefficient because for each of the four direction you only need one comparison, but as you made this iterate in a loop, you actually need to do four comparisons per direction.